tsdb/shard.go

```diff
@@ -610,22 +580,28 @@ func (m *MeasurementFields) CreateFieldIfNotExists(name string, typ influxql.Dat
 	}
 
 	// If we're supposed to limit the number of fields, only 255 are allowed. If we go over that then return an error.
-	if len(m.Fields)+1 > math.MaxUint8 && limitCount {
+	if len(m.fields)+1 > math.MaxUint8 && limitCount {
 		return ErrFieldOverflow
 	}
 
 	// Create and append a new field.
 	f := &Field{
-		ID:   uint8(len(m.Fields) + 1),
+		ID:   uint8(len(m.fields) + 1),
 		Name: name,
 		Type: typ,
 	}
-	m.Fields[name] = f
-	m.Codec = NewFieldCodec(m.Fields)
+	m.fields[name] = f
+	m.Codec = NewFieldCodec(m.fields)
 
 	return nil
 }
 
+func (m *MeasurementFields) Field(name string) *Field {
+	m.mu.RLock()
+	defer m.mu.RUnlock()
```

Comment:
How often is this called? If it's a heavily-used function, maybe read, unlock, then return?